server/src/main/scala/controllers/StaffMovementsPersistence.scala

```diff
@@ -39,10 +39,13 @@ trait StaffMovementsPersistence {
 
     val staffMovementsFuture: Future[Seq[StaffMovement]] = if (forDate.millisSinceEpoch < Crunch.getLocalLastMidnight(SDate.now).millisSinceEpoch) {
       val actorName = "staff-movements-read-actor-" + UUID.randomUUID().toString
-      val staffMovementsReadActor: AskableActorRef = actorSystem.actorOf(Props(classOf[StaffMovementsReadActor], forDate), actorName)
+      val staffMovementsReadActor: ActorRef = actorSystem.actorOf(Props(classOf[StaffMovementsReadActor], forDate), actorName)
 
       staffMovementsReadActor.ask(GetState)
-        .map { case StaffMovements(sm) => sm }.recoverWith { case _ => Future(Seq()) }
+        .map { case StaffMovements(sm) =>
+          staffMovementsReadActor ! PoisonPill
+          sm
+        }.recoverWith { case _ => Future(Seq()) }
```

Comment:
@ryandac should we not also kill the actor on failure?

Reply:
@bnppl yes, changed